x/account/types/genesis_test.go

```diff
@@ -0,0 +1,98 @@
+package types
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestGenesisStateValidate(t *testing.T) {
+	tests := []struct {
+		name     string
+		genState *GenesisState
+		err      error
+	}{
+		{
+			name:     "default is valid",
+			genState: DefaultGenesis(),
+		}, {
+			name: "valid custom genesis",
+			genState: &GenesisState{
+				CoordinatorByAddressList: []*CoordinatorByAddress{
+					{CoordinatorId: 10, Address: "spn1c7gh3kejxm3pzl8fwe65665xncs24x5rl7a8sm"},
+					{CoordinatorId: 11, Address: "spn12330zcy9yez37lzrkm6d7fedcu7hc279sgkh3c"},
+					{CoordinatorId: 12, Address: "spn1d6pd5nk08mu789q4msfpynsuha7yf4wcsvvspr"},
+					{CoordinatorId: 13, Address: "spn1ktzsme3g0ag0236ngvkw62vy9tqrr3xysnhp3g"},
+				},
+				CoordinatorList: []*Coordinator{
+					{CoordinatorId: 10, Address: "spn1c7gh3kejxm3pzl8fwe65665xncs24x5rl7a8sm"},
+					{CoordinatorId: 11, Address: "spn12330zcy9yez37lzrkm6d7fedcu7hc279sgkh3c"},
+					{CoordinatorId: 12, Address: "spn1d6pd5nk08mu789q4msfpynsuha7yf4wcsvvspr"},
+					{CoordinatorId: 13, Address: "spn1ktzsme3g0ag0236ngvkw62vy9tqrr3xysnhp3g"},
+				},
+			},
+			err: nil,
+		}, {
+			name: "duplicated coordinator",
+			genState: &GenesisState{
+				CoordinatorByAddressList: []*CoordinatorByAddress{
+					{CoordinatorId: 10, Address: "spn1c7gh3kejxm3pzl8fwe65665xncs24x5rl7a8sm"},
+					{CoordinatorId: 10, Address: "spn1c7gh3kejxm3pzl8fwe65665xncs24x5rl7a8sm"},
+				},
+			},
+			err: fmt.Errorf("duplicated index for coordinatorByAddress: spn1c7gh3kejxm3pzl8fwe65665xncs24x5rl7a8sm"),
```

Comment:
I'm not strongly against providing the exact error in the assertion, it still make the verification more detailed. Although, I think providing just a boolean `isValid` is sufficient. The main purpose of `Validate` tests is to ensure we don't have false positive that could be really problematic in production

Reply:
I can remove it later, I like to assert the error because we can assert all possible options, and it's clearer for someone who is reading the code. But we can avoid it in the next ones